std/datetime/systime.d

```diff
@@ -11025,32 +11040,42 @@ private:
 /+
     Returns the given hnsecs as an ISO string of fractional seconds.
   +/
-string fracSecsToISOString(int hnsecs) @safe pure nothrow
+string fracSecsToISOString(int hnsecs, int prec = -1) @safe pure nothrow
 {
     import std.array : appender;
     auto w = appender!string();
     try
-        fracSecsToISOString(w, hnsecs);
+        fracSecsToISOString(w, hnsecs, prec);
     catch (Exception e)
         assert(0, "fracSecsToISOString() threw.");
     return w.data;
 }
 
-void fracSecsToISOString(W)(ref W writer, int hnsecs)
+void fracSecsToISOString(W)(ref W writer, int hnsecs, int prec = -1)
 {
     import std.conv : toChars;
     import std.range : padLeft;
 
     assert(hnsecs >= 0);
+    assert(prec >= -1 && prec <= 7);
```

Comment:
This assert should be moved to the user-facing functions. You don't want to have an assert failing inside the innards of a private function, leaving the user with no idea of what he did wrong. I suggest moving this assert to the `toISOExtString` functions and also putting an informative assert message, e.g  `assert(prec >= -1 && prec <=7, "Precision must be in the range [-1, 7]");`